Neos.Neos/Classes/Service/LinkingService.php

```diff
@@ -285,12 +285,12 @@ public function createNodeUri(
         $this->lastLinkedNode = $resolvedNode;
 
         $contentRepository = $this->contentRepositoryRegistry->get($nodeAddress->contentRepositoryId);
-        $workspace = $contentRepository->getWorkspaceFinder()->findOneByName($nodeAddress->workspaceName);
+        $workspace = $contentRepository->findWorkspaceByName($nodeAddress->workspaceName);
 
         $mainRequest = $controllerContext->getRequest()->getMainRequest();
         $uriBuilder = clone $controllerContext->getUriBuilder();
         $uriBuilder->setRequest($mainRequest);
-        $createLiveUri = $workspace && $workspace->isPublicWorkspace() && !$resolvedNode->tags->contain(SubtreeTag::disabled());
+        $createLiveUri = $workspace && $nodeAddress->workspaceName->isLive() && !$resolvedNode->tags->contain(SubtreeTag::disabled());
```

Comment:
This condition seems a bit disjointed now and smells like a conceptual problem. Aka why do I ask the workspaceName if we are live (I know why, but is that sensible?). Anyways nothing we can solve right now, just wanted to point out that it looks a bit strange and the condition read clearer before.